### DISCUSSION TITLE
is it possible to create the following custom plugin?

### DISCUSSION THREAD

**Discussion by hnbdgr1** [WhiteMagic/JoystickGremlin#532] [General]
Hi! 

I'm running into a problem in DCS where i have an instantaneous zoom view bound and use it along with an axis. In some games like Il-2 this works flawlessly as when you release the insta zoom (quick zoom) button, the program reverts to the zoom on the axis automatically. in DCS it's not possible unless you move the axis which is unexpected and jarring when it occurs. 

Is it possible to create a plugin that would do the following behaviour:

On button 1 release -> momentarily press/send current value of axis 1. 

that's it...

I take it natural functionality of joystick gremlin won't allow that hence me asking for a possibility of a plugin. 

I'm happy to try and do it myself with some ai coding help i guess as long as i know it's possible?

thanks!

**Comment by WhiteMagic** [comment]
You shouldn't need a plugin for that. All you would need is a macro action that sets the axis value as desired and has the condition set to "on release".

**Reply by hnbdgr1** [reply]
hmm right. that would work but i'd need a fixed value. how about when i need to store the axis value and recall it later on release, is it possible to use variables that way?

**Reply by WhiteMagic** [reply]
If you need to dynamically read out the value then you have to use a plugin. The UI does not provide any mechanism to do this. This code snippet probably contains the aspects you need to implement that functionality though.

https://gist.github.com/WhiteMagic/f6dab456dc0cc3b7496b768ab735f4ba
